handlers/oam_handler.go

```diff
@@ -168,10 +168,22 @@ func (h *Handler) POSTOAMRegisterHandler(typ string, r *http.Request) error {
 	}
 
 	if typ == "workload" {
-		return core.RegisterWorkload(body)
+		err := core.RegisterWorkload(body)
```

Comment:
Why duplicate if you want to run it in every case? I can see it is not being replicated when registering the scopes, that is not desirable though.

Reply:
was confused here , about if we wanted it to trigger for scopes too. Will make the change